cmake/modules/autopas_auto4omp.cmake

```diff
@@ -0,0 +1,576 @@
+set(AUTOPAS_AUTO4OMP_GIT_TAG "master" CACHE STRING "${AUTOPAS_AUTO4OMP_GIT_TAG_DOC}")
+set(AUTOPAS_AUTO4OMP_INSTALL_DIR "" CACHE PATH "${AUTOPAS_AUTO4OMP_INSTALL_DIR_DOC}")
+set(AUTOPAS_NVCC_GNUC_PATH OFF CACHE FILEPATH "${AUTOPAS_NVCC_GNUC_PATH_DOC}")
+set(AUTOPAS_LLVM_LIT_EXECUTABLE OFF CACHE FILEPATH "${AUTOPAS_LLVM_LIT_EXECUTABLE_DOC}")
+set(AUTOPAS_FILECHECK_EXECUTABLE OFF CACHE FILEPATH "${AUTOPAS_FILECHECK_EXECUTABLE_DOC}")
+set(AUTOPAS_pthread_LIBRARY OFF CACHE FILEPATH "${AUTOPAS_pthread_LIBRARY_DOC}")
+
+# Version settings. If AutoPas or Auto4OMP support new packages, update here.
+set(AUTOPAS_NVCC_MAX_GCC 12 CACHE STRING ${AUTOPAS_NVCC_MAX_GCC_DOC})
+set(AUTOPAS_OMP_VERSION 45 CACHE STRING ${AUTOPAS_OMP_VERSION_DOC})
+
+if (NOT AUTOPAS_AUTO4OMP)
+    # If Auto4OMP disabled, notify.
+    message(STATUS "Auto4OMP disabled.")
+    set(AUTOPAS_LB4OMP OFF CACHE BOOL ${AUTOPAS_LB4OMP_DOC} FORCE)
+elseif (NOT AUTOPAS_OPENMP)
+    # If OpenMP disabled, warn.
+    message(WARNING "OpenMP must be enabled to use Auto4. Auto4OMP disabled.")
+    set(AUTOPAS_AUTO4OMP OFF CACHE BOOL ${AUTOPAS_AUTO4OMP_DOC} FORCE)
+    set(AUTOPAS_LB4OMP OFF CACHE BOOL ${AUTOPAS_LB4OMP_DOC} FORCE)
+else ()
+    # Notify.
+    message(STATUS "Auto4OMP enabled.")
+
+    if (NOT CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+        # If Clang isn't used, warn. Auto4OMP is tested with up to clang 7 and gcc 4. Modern gcc fails.
+        message(WARNING "Auto4OMP needs clang, but ${CMAKE_CXX_COMPILER_ID} is used. Building may produce errors.")
+    endif ()
+
+    # Auto4OMP's CMake variables:
+
+    ## OpenMP version:
+    ### AutoPas supports up to OpenMP 4.5. Pass this to Auto4OMP.
+    set(LIBOMP_OMP_VERSION ${AUTOPAS_OMP_VERSION} CACHE INTERNAL "OpenMP version for Auto4OMP to build for.")
+
+    ### OpenMP tools is only supported from OpenMP 5.
+    if (${AUTOPAS_OMP_VERSION} GREATER_EQUAL 50)
+        set(LIBOMP_OMPT_SUPPORT ON CACHE INTERNAL "OpenMP tools support.")
+    else ()
+        set(LIBOMP_OMPT_SUPPORT OFF CACHE INTERNAL "OpenMP tools support.")
+    endif ()
+
+    ## High resolution timer:
+    ### Auto4OMP needs a high resolution timer to make measurements; either a cycle counter or time-stamp counter.
+
+    ### Cycle counter:
+    #### __builtin_readcyclecounter() is available since clang 4 (possibly prior). [1]
+    #### On systems lacking a cycle counter register or similar, the function defaults to 0.
+    #### Compile and run CycleCounterQuery.cpp to check if a cycle counter's available, inspired from [2, 3].
+    execute_process(
+            COMMAND mkdir -p ${CMAKE_BINARY_DIR}/scripts
+            COMMAND ${CMAKE_CXX_COMPILER} ${CMAKE_SOURCE_DIR}/src/autopas/utils/CycleCounterQuery.cpp
+            -o ${CMAKE_BINARY_DIR}/scripts/has_builtin_readcyclecounter
+    )
+    execute_process(
+            COMMAND "${CMAKE_BINARY_DIR}/scripts/has_builtin_readcyclecounter"
+            OUTPUT_VARIABLE HAS_BUILTIN_READCYCLECOUNTER_OUTPUT
+    )
```

Comment:
Since the code's simple, I piped it into the compiler directly from the CMake module. It still compiles and runs with execute_process(), but there's no need for the cpp anymore, so I'm marking this resolvedbut there's no need for the cpp anymore, so I'm marking this resolved for now. for now.